src/processTargets/modifiers/scopeHandlers/OneOfScopeHandler.ts

```diff
@@ -0,0 +1,113 @@
+import { minBy } from "lodash";
+import { Position, Range, TextEditor } from "vscode";
+import { getScopeHandler } from ".";
+import {
+  Direction,
+  OneOfScopeType,
+} from "../../../typings/targetDescriptor.types";
+import NotHierarchicalScopeError from "./NotHierarchicalScopeError";
+import type { TargetScope } from "./scope.types";
+import { ScopeHandler } from "./scopeHandler.types";
+
+export default class OneOfScopeHandler implements ScopeHandler {
+  private scopeHandlers: ScopeHandler[] = this.scopeType.scopeTypes.map(
+    (scopeType) => {
+      const handler = getScopeHandler(scopeType, this.languageId);
+      if (handler == null) {
+        throw new Error(`No available scope handler for '${scopeType.type}'`);
+      }
+      return handler;
+    }
+  );
+
+  public iterationScopeType: OneOfScopeType = {
+    type: "oneOf",
+    scopeTypes: this.scopeHandlers.map(
+      ({ iterationScopeType }) => iterationScopeType
+    ),
+  };
```

Comment:
I'm worried I introduced an infinite recursion here. Should be able to work around it by making it lazy if necessary